src/Spelling/WordParser.cs

```diff
@@ -0,0 +1,566 @@
+        /// </summary>
+        /// <returns>
+        /// The System.String.
+        /// </returns>
+        public string PeekWord()
+        {
+            return _peekedWord ?? (_peekedWord = NextWordCore());
+        }
+
+        private static bool IsDigit(char c)
+        {
+            return char.IsDigit(c);
+        }
+
+        private static bool IsHexDigit(char c)
+        {
+            switch (c)
+            {
+                case 'A':
+                case 'B':
+                case 'C':
+                case 'D':
+                case 'E':
+                case 'F':
+                case 'a':
+                case 'b':
+                case 'c':
+                case 'd':
+                case 'e':
+                case 'f':
+                    return true;
+            }
+
+            return IsDigit(c);
+        }
+
+        private static bool IsIntraWordPunctuation(char c)
+        {
+            switch (c)
+            {
+                case '\x00ad':
+                case '’':
+                case '\'':
+                case '-':
+                    return true;
+            }
+
+            return false;
+        }
+
+        private static bool IsLetterOrDigit(char c)
+        {
+            return char.IsLetterOrDigit(c);
+        }
+
+        private static bool IsLetterWithoutCase(char c)
+        {
+            return (char.IsLetter(c) && !char.IsUpper(c)) && !char.IsLower(c);
+        }
+
+        private static bool IsLower(char c)
+        {
+            return char.IsLower(c);
+        }
+
+        private static bool IsUpper(char c)
+        {
+            return char.IsUpper(c);
+        }
+
+        private bool IsIgnored(char c)
+        {
+            if (!SkipMnemonics)
+            {
+                return false;
+            }
+
+            if (c != '&')
+            {
+                return c == '_';
+            }
+
+            return true;
+        }
+
+        private string NextWordCore()
+        {
+            _buffer.Length = 0;
+            if (ParseNext())
+            {
+                return _buffer.ToString();
+            }
+
+            return null;
+        }
+
+        private void ParseAllCaps()
+        {
+            char ch;
+            do
+            {
+                Read();
+                ch = Peek();
+            }
+            while (IsUpper(ch));
+            if (ch == 's')
+            {
+                Read();
+                ch = Peek();
+            }
+
+            while (IsLower(ch))
+            {
+                Unread();
+                ch = Peek();
+            }
+        }
+
+        private void ParseHex()
+        {
+            do
+            {
+                Read();
+            }
+            while (IsHexDigit(Peek()));
+        }
+
+        private void ParseInteger()
+        {
+            do
+            {
+                Read();
+            }
+            while (IsDigit(Peek()));
+        }
+
+        private void ParseLatex()
```

Comment:
❔ Why? LaTeX isn't supported in C# code comments.

Reply:
No longer applies